controllers/operandregistry_controller.go

```diff
@@ -72,37 +69,19 @@ func (r *OperandRegistryReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 
 		return ctrl.Result{}, err
 	}
-	// Check if all the catalog sources ready for deployment
-	isReady, err := r.checkCatalogSourceStatus(instance)
-	if err != nil {
-		klog.Errorf("failed to check the CatalogSource status for OperandRegistry %s/%s : %v", req.NamespacedName.Namespace, req.NamespacedName.Name, err)
-		return ctrl.Result{}, err
-	}
 
-	if isReady {
-		if instance.Status.OperatorsStatus == nil || len(instance.Status.OperatorsStatus) == 0 {
-			instance.UpdateRegistryPhase(operatorv1alpha1.RegistryReady)
-		} else {
-			instance.UpdateRegistryPhase(operatorv1alpha1.RegistryRunning)
-		}
-		if err := r.updateOperandRegistryStatus(instance); err != nil {
-			klog.Errorf("failed to update the status for OperandRegistry %s/%s : %v", req.NamespacedName.Namespace, req.NamespacedName.Name, err)
-			return ctrl.Result{}, err
-		}
+	if instance.Status.OperatorsStatus == nil || len(instance.Status.OperatorsStatus) == 0 {
+		instance.UpdateRegistryPhase(operatorv1alpha1.RegistryReady)
 	} else {
-		instance.UpdateRegistryPhase(operatorv1alpha1.RegistryWaiting)
-		if err := r.updateOperandRegistryStatus(instance); err != nil {
-			klog.Errorf("failed to update the status for OperandRegistry %s/%s : %v", req.NamespacedName.Namespace, req.NamespacedName.Name, err)
-			return ctrl.Result{}, err
-		}
-		// When catalog source not ready, reconcile every 1 minute to check the status
-		klog.V(2).Info("Waiting for all catalog source ready ...")
-		return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+		instance.UpdateRegistryPhase(operatorv1alpha1.RegistryRunning)
+	}
+	if err := r.updateOperandRegistryStatus(instance); err != nil {
```

Comment:
In this method `updateOperandRegistryStatus`,  we use the method `wait.PollImmediate` to update status, it will re-fetch operandregistry, before change in method `updateRegistryOperatorsStatus` will be lost, you can combine following logic in one method
```
	if err := r.updateRegistryOperatorsStatus(instance); err != nil {
		klog.Errorf("failed to update the status for OperandRegistry %s/%s : %v", req.NamespacedName.Namespace, req.NamespacedName.Name, err)

		return ctrl.Result{}, err
	}

	if instance.Status.OperatorsStatus == nil || len(instance.Status.OperatorsStatus) == 0 {
		instance.UpdateRegistryPhase(operatorv1alpha1.RegistryReady)
	} else {
		instance.UpdateRegistryPhase(operatorv1alpha1.RegistryRunning)
	}
	if err := r.updateOperandRegistryStatus(instance); err != nil {
		klog.Errorf("failed to update the status for OperandRegistry %s/%s : %v", req.NamespacedName.Namespace, req.NamespacedName.Name, err)
		return ctrl.Result{}, err
	}
```

Reply:
I see. Since our registry logic become simpler, could we just drop the `wait.PollImmediate` method and using the `reconcile.Result` to manage the requeue.

Reply:
I believe the change wouldn't be lost. The changed instance is inputted by parameter `newRegistryInstance`. During every time's retry, the function doesn't update the `newRegistryInstance`, but deep copy its status and copy to the current instance.

Reply:
you correct.